tests/js/server/paths/filesystem.js

```diff
@@ -0,0 +1,325 @@
+    let rc = fs.isDirectory(fn);
+    assertEqual(rc, isDirectory, 'Expected ' + fn + ' to be a ' + isDirectory ? "directory.":"file.");
+  }
+
+  function tryCreateDirectoryAllowed(fn) {
+    let rc = fs.makeDirectory(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, true);
+  }
+
+  function tryRemoveDirectoryAllowed(fn) {
+    let rc = fs.removeDirectory(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, false);
+  }
+
+  function tryCreateDirectoryRecursiveAllowed(fn) {
+    let rc = fs.makeDirectoryRecursive(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, true);
+  }
+
+  function tryRemoveDirectoryRecursiveAllowed(fn) {
+    let rc = fs.removeDirectoryRecursive(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to be removed.');
+    tryIsDirectoryAllowed(fn, false);
+  }
+
+  function tryIsFileAllowed(fn, isFile) {
+    let rc = fs.isFile(fn);
+    assertEqual(rc, isFile, 'Expected ' + fn + ' to be a ' + isFile ? "file.":"directory.");
+  }
+
+  function tryListFileAllowed(dn, expectCount) {
+    let rc = fs.list(dn);
+    assertEqual(rc.length, expectCount, 'Expected ' + dn + ' to contain ' + expectCount + " files.");
+  }
+
+  function tryListTreeAllowed(dn, expectCount) {
+    let rc = fs.listTree(dn);
+    assertEqual(rc.length, expectCount, 'Expected ' + dn + ' to contain ' + expectCount + " files.");
+  }
+
+  function tryGetTempFileAllowed(dn) {
+    let tfn = fs.getTempFile();
+    fs.write(tfn, "hello");
+    assertTrue(fs.isFile(tfn));
+    fs.remove(tfn);
+  }
+
+  function tryMakeAbsoluteAllowed(fn) {
+    fs.makeAbsolute(fn);
+  }
+  function tryCopyFileAllowed(sn, tn) {
+    fs.copyFile(sn, tn);
+    tryExistsAllowed(sn, true);
+    tryExistsAllowed(tn, true);
+  }
+  function tryCopyRecursiveFileAllowed(sn, tn) {
+    fs.copyRecursive(sn, tn);
+    tryExistsAllowed(sn, true);
+    tryExistsAllowed(tn, true);
+  }
+  function tryMoveFileAllowed(sn, tn) {
+    fs.move(sn, tn);
+    tryExistsAllowed(sn, false);
+    tryExistsAllowed(tn, true);
+  }
+  function tryRemoveFileAllowed(fn) {
+    fs.remove(fn);
+    tryExistsAllowed(fn, false);
+  }
+  function tryLinkFileAllowed(sn, tn) {
+    fs.linkFile(sn, tn);
+    tryExistsAllowed(sn, true);
+    tryExistsAllowed(tn, true);
+  }
+
+  function tryZipFileAllowed(zip, sourceDir) {
+    let files = [];
+    try {
+      files = fs.list(sourceDir).filter(function (fn) {
+        if (fn === 'into_forbidden.txt') {
+          return false;
+        }
+        return fs.isFile(fs.join(sourceDir, fn));
+      });
+      fs.zipFile(zip, sourceDir, files);
+    } catch (err) {
+      assertTrue(false, "failed to Zip into " + zip + " these files: " + sourceDir + "[ " + files + " ] - " + err);
+    }
+    tryExistsAllowed(zip, true);
+  }
+
+  function tryUnZipFileAllowed(zip, sn) {
+    try {
```

Comment:
These try...catch wrappers just distract from the source of error.
Whenever one of the file operations fails, it can clearly be figured out from the stack trace which file/test caused the error.
Adding an extra framework around the assertions of the test framework is just clutter IMHO.